core/psh/ps/ps.c

```diff
@@ -133,9 +155,9 @@ int psh_ps(int argc, char **argv)
 		printf("%6s ", buff);
 
 		if (!threads)
-			printf("%3u %-28s\n", info[i].tid, info[i].name);
+			printf("%3u %-*s\n", info[i].tid, (fullcmd ? sizeof(info[i].name) : 28), info[i].name);
 		else
-			printf("%-32s\n", info[i].name);
+			printf("%-*s\n", (fullcmd ? sizeof(info[i].name) : 32), info[i].name);
```

Comment:
I'd replace `sizeof(info[i].name)` with `(int)strlen(info[i].name)`. This way we can avoid empty lines (current behaviour with printing 128 cmd characters looks like this):
![ps-f](https://user-images.githubusercontent.com/8835317/120317065-7831f400-c2de-11eb-88b8-cfcbe06dfaa6.png)

Reply:
Sorry, my bad, as this is the last field I'm tempted to use precision instead of field width (so write `%.*s`) instead. I'll test it in a moment (as per specs - `the maximum number of characters to be printed from a string`). I think we don't need expansion field expansion with spaces in this case

Reply:
Using precision instead of field width solves the newline problem on my setup (80x25 terminal):
![image](https://user-images.githubusercontent.com/3939983/120323683-09589900-c2e6-11eb-9811-9c73cd33f4ba.png)

I believe original `ps` actually trims the result to current terminal size which would probably be better than current solution, but there are some problems with terminal width on our targets so it's not applicable right away (eg. https://github.com/phoenix-rtos/phoenix-rtos-devices/issues/102)